### PR TITLE
Feature/238-association user and financial transaction category

### PR DESCRIPTION
**Pull request by SzanieckiFranciszek**
#### Checklist template for LEADERS

&#x2611; | &#x2610;

&#x2610; New code is covered with tests and coverage requirements are met.
&#x2610; Build is not failing.
&#x2610; All tests are passing.
&#x2610; All conflicts are resolved and the branch is rebased on top of the up-to-date parent branch.
&#x2610; All issue requirements are implemented.
&#x2610; At least two reviewers are assigned.
&#x2610; The Pull request is linked to the related issue.
&#x2610; Correct branch name. Convention [https://github.com/bycza-zagroda/track-expenses-app-documentation/blob/develop/githubwork/GITHUB_WORK.md]
&#x2610; Correct unit test name - suffix 'Test' in the end.
&#x2610; Correct integration test name - suffix 'IT' in the end.
&#x2610; Positive and negative test paths have been implemented.

### CODE REVIEW

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/model/FinancialTransactionCategory.java [hunk 1]
```diff
@@ -74,4 +80,18 @@ public FinancialTransactionCategory(String name, FinancialTransactionType type)
...
+    }
+
+    public void addUser(User user){
+        this.setUser(user);
+    }
+
+    public void removeUser(){
+        this.setUser(null);
+    }
 }
```
Comment (Iwona007) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/model/FinancialTransactionCategory.java:97]: Please remove lines from 90 - 96 and add these one: 
    public void addFinancialTransactionCategory(FinancialTransactionCategory financialTransactionCategory){
        this.financialTransactionCategories.add(financialTransactionCategory);
        financialTransactionCategory.setUser(this);
    }

    public void removeFinancialTransactionCategory(FinancialTransactionCategory financialTransactionCategory) {
        this.financialTransactionCategories.remove(financialTransactionCategory);
        financialTransactionCategory.setUser(null);
    }

**Review by Iwona007 — CHANGES_REQUESTED** [review]
Changes are required

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/model/User.java [hunk 2]
```diff
@@ -67,6 +67,11 @@ public class User implements Serializable {
     @Column(name = "user_status", columnDefinition = "ENUM('VERIFIED', 'UNVERIFIED', 'BLOCKED', 'BANNED')")
     private UserStatus userStatus;
 
+    @OneToMany(mappedBy = "user",
+            cascade = CascadeType.ALL,
+            orphanRemoval = true)
+    private List<FinancialTransactionCategory> financialTransactionCategories = new ArrayList<>();
+
     public void addWallet(Wallet wallet) {
```
Comment (Iwona007) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/model/User.java:75]: Please add:
public void addFinancialTransactionCategory(FinancialTransactionCategory financialTransactionCategory){
this.financialTransactionCategories.add(financialTransactionCategory);
financialTransactionCategory.setUser(this);
}

public void removeFinancialTransactionCategory(FinancialTransactionCategory financialTransactionCategory) {
    this.financialTransactionCategories.remove(financialTransactionCategory);
    financialTransactionCategory.setUser(null);
}

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/model/FinancialTransactionCategory.java (hunk 1 above)
Comment (SzanieckiFranciszek) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/model/FinancialTransactionCategory.java:97]: Added this method to User class, according to our findings.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/model/User.java (hunk 2 above)
Comment (SzanieckiFranciszek) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/model/User.java:75]: Done.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/model/FinancialTransactionCategory.java [hunk 3]
```diff
@@ -74,4 +80,18 @@ public FinancialTransactionCategory(String name, FinancialTransactionType type)
         this.creationDate = Instant.now();
     }
 
+    public FinancialTransactionCategory(String name, FinancialTransactionType type, User user) {
+        this.name = name;
+        this.type = type;
+        this.creationDate = Instant.now();
+        this.user = user;
+    }
+
```
[the remark this answers is not in the document]
Comment (SzanieckiFranciszek) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/model/FinancialTransactionCategory.java:89]: Done.

**Review by Iwona007 — APPROVED** [review]
Approve

**Pull request merged**